src/main/java/io/cdap/plugin/spark/dynamic/ScalaSparkSink.java

```diff
@@ -109,11 +116,16 @@ public void run(SparkExecutionPluginContext context, JavaRDD<StructuredRecord> j
     @Nullable
     private final Boolean deployCompile;
 
+    @Name("referenceName")
+    @Description("This will be used to uniquely identify this source/sink for lineage, annotating metadata, etc.")
+    public String referenceName;
```

Comment:
You will need to check add the config to widget file also, check https://github.com/cdapio/hydrator-plugins/blob/develop/core-plugins/widgets/File-batchsource.json as an example